### PR TITLE
Remove line about combining headers as an optimization

### PR DESCRIPTION
**Pull request by easwars**
The removed line is factually incorrect. To be factually correct, it should have said something like "gRPC optimizes for the unary case by combining headers, data and trailer into a single TCP packet". But even that cannot be always guaranteed. So, removing this line altogether from what is a beginner guide on metadata, seems more appropriate.

### CODE REVIEW

File: content/en/docs/guides/metadata.md [hunk 1]
```diff
@@ -116,4 +115,4 @@ the server.
 
 [Python Metadata]: https://github.com/grpc/grpc/tree/master/examples/python/metadata
 
-[Go Documentation]: https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md
```
Comment (dfawley) [inline content/en/docs/guides/metadata.md:119]: Why is this line changed? Did you delete or add a newline at the end of the file or something?  Github is hiding it from me.  (Please make sure there _is_ a newline.)

**Review by dfawley — APPROVED** [review]

File: content/en/docs/guides/metadata.md (hunk 1 above)
Comment (easwars) [reply content/en/docs/guides/metadata.md:119]: Crap, looks like my editor is doing this by itself. What purpose does the newline at the end serve?

Comment (easwars) [reply content/en/docs/guides/metadata.md]: Looks like there was no newline earlier at the end of the last line of the doc, and my editor was adding it. Got rid of it now.

Comment (dfawley) [reply content/en/docs/guides/metadata.md]: Oh.. OK.  Anything that cares (that I've ever seen) always _wants_ there to be a newline, but not changing it SGTM.
